freqtrade/rpc/rpc.py

```diff
@@ -1115,31 +1115,72 @@ def _rpc_delete(self, trade_id: int) -> dict[str, str | int]:
                 "cancel_order_count": c_count,
             }
 
-    def _rpc_list_custom_data(self, trade_id: int, key: str | None) -> list[dict[str, Any]]:
-        # Query for trade
-        trade = Trade.get_trades(trade_filter=[Trade.id == trade_id]).first()
-        if trade is None:
-            return []
-        # Query custom_data
-        custom_data = []
-        if key:
-            data = trade.get_custom_data(key=key)
-            if data:
-                custom_data = [data]
+    def _rpc_list_custom_data(
+        self, trade_id: int | None = None, key: str | None = None, limit: int = 100, offset: int = 0
+    ) -> list[dict[str, Any]]:
+        """
+        Fetch custom data for a specific trade, or all open trades if `trade_id` is not provided.
+        Pagination is applied via `limit` and `offset`.
+
+        Returns an array of dictionaries, each containing:
+        - "trade_id": the ID of the trade (int)
+        - "custom_data": a list of custom data dicts, each with the fields:
+                "id", "ft_trade_id", "cd_key", "cd_type", "cd_value", "created_at", "updated_at"
+        """
+        trades: Sequence[Trade]
+        if trade_id is None:
+            # Get all open trades
+            order_by: Any = Trade.close_date.desc()
```

Comment:
```suggestion
            order_by: Any = Trade.open_date.desc()
```
Open date? `close_date` will be empty for all open trades.
While unlikely (you'd have to have open trades > limit) - a low "limit" number could still potentially break pagination.